### ISSUE TITLE
Pressing q necessary to continue when plot2d output to svg

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2511]
Imported from SourceForge on 2024-07-06 10:07:42
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-03-04 03:17:54
Original: https://sourceforge.net/p/maxima/bugs/3953

---

Currently (ac71e89) plot2d emits `pause mouse close` at the end of Gnuplot output. When the output is going to a file, e.g. svg, it's necessary to press q to continue, even though there is no visible output. I didn't try any other output formats (png, pdf, etc., if any) although I would guess they are all the same.

I think if the output is going to a file (or any other invisible destination, I guess), then `pause mouse close` should be suppressed.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:07:44
Created by **[villate](https://sourceforge.net/u/villate/)** on 2022-03-04 11:52:40
Original: https://sourceforge.net/p/maxima/bugs/3953/#b28c

---

"pause mouse close" is only emitted when the output is not sent to a file. I have checked that commands such as:
~~~
plot2d(x^2,[x,-2,2],[pdf_file,"./test.pdf"])
~~~
create a file as expected. Can you please show us the command you used and which required you to press q before creating the file?

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:07:47
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-03-05 06:27:25
Original: https://sourceforge.net/p/maxima/bugs/3953/#e78d

---

OK, I tried some different combinations. I'm working on Linux with a current build (4ea001).

 * `set_plot_option([plot_format, gnuplot]);` (didn't try other options)
 *  `set_plot_option([gnuplot_term, "x11 persist font 'urw palladio l,24'"]);` (I assume this has no effect, including it just in case)
 *  tried Clisp, SBCL, ECL, Clozure CL, CMUCL. 
 *  tried svg output, e.g. `plot2d(sin(x), [x, 0, 1], [svg_file, "foo.svg"]);`
 
 The pause for `q<return>` happens only with Clisp. For the other Lisps, there is no pause; Maxima prints the next label and reads Maxima input immediately. However, `pause mouse close` is output at the end of the Gnuplot output file for all Lisps.
 
 Looking at src/gnuplot_def.lisp, I see `pause mouse close` is only output when `plot_format` is `gnuplot`, and it is output unconditionally (doesn't depend on whether there is an output file or screen display). 
 
 For the record, I launched gnuplot and entered `load "maxoutnnnn.gnuplot"`,  where the maxout file is one generated for svg output above. Gnuplot pauses for `q<return>` even though output is going to a file.
 
So in summary, I think if the intent is that `pause mouse close` is only output for visible displays, it's not working as intended.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:07:51
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-03-05 16:50:29
Original: https://sourceforge.net/p/maxima/bugs/3953/#131c

---

PS. Gnuplot reports its version as "Version 5.4 patchlevel 2    last modified 2021-06-01".

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:07:54
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-03-05 17:20:08
Original: https://sourceforge.net/p/maxima/bugs/3953/#1c1c

---

PPS. I find that running gnuplot without a controlling terminal (i.e. via `nohup gnuplot`) causes gnuplot to run to completion and generate the svg output without waiting for `q<return>`. Just to be clear, where maxoutnnn.gnuplot outputs to svg and has `pause mouse close` at the end:

 * `gnuplot maxoutnnnn.gnuplot` displays nothing, prints nothing, and waits until `q<return>` before closing. Svg is created successfully.
 *  `nohup gnuplot maxoutnnnn.gnuplot` same as above, except it does not wait for `q<return>`.

This difference of behavior in gnuplot might explain the per-Lisp behavior described above -- the Lisp implementation might or might not supply an input stream to a program launched via `$SYSTEM`.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:07:57
Created by **[villate](https://sourceforge.net/u/villate/)** on 2022-03-09 12:47:28
Original: https://sourceforge.net/p/maxima/bugs/3953/#5a65

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 10:08:01
Created by **[villate](https://sourceforge.net/u/villate/)** on 2022-03-09 12:47:28
Original: https://sourceforge.net/p/maxima/bugs/3953/#1497

---

Fixed with commit [900105]
